### PR TITLE
[FGA-73] Add FGA support

### PR DESCRIPTION
**Pull request by atainter**
## Description
Adding support for FGA APIs

Adds the following methods to an `FGA` submodule in the SDK:

```python
    def get_resource(self, resource_type: str, resource_id: str) -> Resource: ...

    def list_resources(
        self,
        resource_type: Optional[str] = None,
        search: Optional[str] = None,
        limit: int = DEFAULT_RESPONSE_LIMIT,
        order: PaginationOrder = Order.Desc.value,
        before: Optional[str] = None,
        after: Optional[str] = None,
    ) -> ResourceListResource: ...

    def create_resource(
        self,
        resource_type: str,
        resource_id: str,
        meta: Dict[str, Any],
    ) -> Resource: ...

    def update_resource(
        self,
        resource_type: str,
        resource_id: str,
        meta: Dict[str, Any],
    ) -> Resource: ...

    def delete_resource(self, resource_type: str, resource_id: str) -> None: ...

    def list_resource_types(
        self,
        limit: int = DEFAULT_RESPONSE_LIMIT,
        order: PaginationOrder = Order.Desc.value,
        before: Optional[str] = None,
        after: Optional[str] = None,
    ) -> WorkOsListResource[ResourceType, ListArgs, ListMetadata]: ...

    def list_warrants(
        self,
        resource_type: Optional[str] = None,
        resource_id: Optional[str] = None,
        relation: Optional[str] = None,
        subject_type: Optional[str] = None,
        subject_id: Optional[str] = None,
        subject_relation: Optional[str] = None,
        limit: int = DEFAULT_RESPONSE_LIMIT,
        order: PaginationOrder = Order.Desc.value,
        before: Optional[str] = None,
        after: Optional[str] = None,
        warrant_token: Optional[str] = None,
    ) -> WorkOsListResource[Warrant, WarrantListFilters, ListMetadata]: ...

    def write_warrant(
        self,
        op: WarrantWriteOperation,
        resource_type: str,
        resource_id: str,
        relation: str,
        subject_type: str,
        subject_id: str,
        subject_relation: Optional[str] = None,
        policy: Optional[str] = None,
    ) -> WriteWarrantResponse: ...

    def batch_write_warrants(
        self, batch: List[WarrantWrite]
    ) -> WriteWarrantResponse: ...

    def check(
        self,
        checks: List[WarrantCheck],
        op: Optional[CheckOperation] = None,
        debug: bool = False,
        warrant_token: Optional[str] = None,
    ) -> CheckResponse: ...
```

Remaining endpoints will be added in a future PR.

### CODE REVIEW

File: workos/types/fga/resource_types.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import Any, Dict, Optional
+
+from workos.types.workos_model import WorkOSModel
+
+
+class ResourceType(WorkOSModel):
+    type: str
+    relations: Dict[str, Any]
+    created_at: Optional[str] = None
```
Comment (mattgd) [inline workos/types/fga/resource_types.py:9]: Confirming `created_at` can in fact be None?

File: workos/fga.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+class FGAModule(Protocol):
+    def get_resource(self, resource_type: str, resource_id: str) -> Resource: ...
+
+    def list_resources(
+        self,
+        resource_type: Optional[str] = None,
+        search: Optional[str] = None,
+        limit: int = DEFAULT_RESPONSE_LIMIT,
+        order: PaginationOrder = Order.Desc.value,
```
Comment (mattgd) [inline workos/fga.py:50]: Made me realize we actually meant to remove Order. Could just use `PaginationOrder = "desc"` here.

File: workos/fga.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+        resource_id: str,
+        meta: Dict[str, Any],
+    ) -> Resource: ...
+
+    def delete_resource(self, resource_type: str, resource_id: str) -> None: ...
+
+    def list_resource_types(
+        self,
+        limit: int = DEFAULT_RESPONSE_LIMIT,
+        order: PaginationOrder = Order.Desc.value,
```
Comment (mattgd) [inline workos/fga.py:74]: Same re: Order

File: tests/test_fga.py [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+                                    "inherit_if": "member",
+                                    "of_type": "tenant",
+                                    "with_relation": "member",
+                                },
+                            ],
+                        }
+                    },
+                },
+            ],
+            "list_metadata": {"after": "{{ orderASCLimit3.list_metadata.after }}"},
```
Comment (mattgd) [inline tests/test_fga.py:286]: Is this the intended format of `list_metadata["after"]`?
```
{{ orderASCLimit3.list_metadata.after }}
```

**Review by mattgd — APPROVED** [review]
Only a few minor comments, but looks good!

File: workos/types/fga/resource_types.py (hunk 1 above)
Comment (atainter) [reply workos/types/fga/resource_types.py:9]: Confirmed. I believe this is an issue with the API that will be resolved soon.

File: tests/test_fga.py (hunk 4 above)
Comment (atainter) [reply tests/test_fga.py:286]: It is not - I copied these responses from some API tests and missed this part. Will update.

**Review by cadef — APPROVED** [review]

**Pull request merged**